src/System/TimedScopes/TimedScopeEventSource.cs

```diff
@@ -0,0 +1,233 @@
+		[NonEvent]
+		public void LogEvent(
+			Category logCategory,
+			string name,
+			string subtype,
+			string metadata,
+			string serviceName,
+
+			TimedScopeResult result,
+			string correlationId,
+			double durationMs
+		)
+		{
+			string nameAsString = SanitizeMdmString(name, nameof(name), name, logCategory);
+			string subTypeAsString = SanitizeMdmString(subtype, nameof(subtype), name, logCategory);
+			string metaDataAsString = SanitizeMdmString(metadata, nameof(metadata), name, logCategory);
+			string serviceNameAsString = SanitizeMdmString(serviceName, nameof(serviceName), name, logCategory);
+			string correlationIdAsString = SanitizeMdmString(correlationId, nameof(correlationId), name, logCategory);
+
+			WriteTimedScopeTrxEvent(
+				nameAsString,
+				subTypeAsString,
+				metaDataAsString,
+				serviceNameAsString,
+
+				logCategory.Name,
+				result.ToString(),
+				correlationIdAsString,
+				Convert.ToInt64(durationMs));
+		}
+
+
+		private static string SanitizeMdmString(string mdmString, string name, string activityName, Category logCategory)
+		{
+			string validatedString = Convert.ToString(mdmString) ?? string.Empty;
+
+			if (validatedString.Length > 1024)
+			{
+				ULSLogging.LogTraceTag(0x23857681 /* tag_97x0b */, logCategory, Levels.Warning, MdmStringLimitMessage, 1024, name, activityName, validatedString.Length);
+				validatedString = validatedString.Substring(0, 1024);
+			}
+
+			return validatedString;
+		}
+
+
+		private const string MdmStringLimitMessage =
+			"MDM enforces a string length limit of {0} characters per dimension. Truncating length of dimension {1} on activity {2} from {3} chars in order to allow upload of the metric";
+
+		#endregion
+
+
+		#region Events
+
+		/// <summary>
+		/// Writes a Timed scope event with Level=Informational.
+		/// </summary>
+		/// <param name="name">Timed scope name</param>
+		/// <param name="subType">Subtype</param>
+		/// <param name="metadata">Metadata</param>
+		/// <param name="serviceName">Service name</param>
+		/// <param name="userHash">User hash</param>
+		/// <param name="logCategory">Log category</param>
+		/// <param name="result">Result</param>
+		/// <param name="correlationId">Correlation Id</param>
+		/// <param name="durationMs">Duration in ms</param>
+		[Event((int)EventIds.LogTimedScopeEventId, Level = EventLevel.Informational, Version = 3)]
+		private void WriteTimedScopeEvent(
+			string name,
+			string subType,
+			string metadata,
+			string userHash,
+			string serviceName,
+			string logCategory,
+			string result,
+			string correlationId,
+			long durationMs)
+		{
+			WriteEvent((int)EventIds.LogTimedScopeEventId, name ?? string.Empty, subType ?? string.Empty, metadata ?? string.Empty, userHash ?? string.Empty, serviceName ?? string.Empty, logCategory, result, correlationId, durationMs);
+		}
+
+
+		/// <summary>
+		/// Writes a Timed scope trace event with Level=Informational.
```

Comment:
>trace [](start = 27, length = 5)

What exactly does trace mean in this context? The method name includes `Trx` which potentially suggests a transaction. It would be good to clarify this for external consumers

Reply:
I think trace event here refers to ETW (Event Tracing for Windows) and it's just the name of log event. I agree that `Trx` could be expanded, but I don't want to change the contract due to the reasons stated [here](https://github.com/microsoft/Omex/pull/151/files#r356214797)